pkgs/development/libraries/xsimd/default.nix

```diff
@@ -1,35 +1,34 @@
-{ lib, stdenv, fetchFromGitHub, cmake, gtest }:
+{ lib, stdenv, fetchFromGitHub, cmake, doctest }:
 stdenv.mkDerivation rec {
   pname = "xsimd";
-  version = "9.0.1";
+  version = "10.0.0";
   src = fetchFromGitHub {
     owner = "xtensor-stack";
     repo = "xsimd";
     rev = version;
-    sha256 = "sha256-onALN6agtrHWigtFlCeefD9CiRZI4Y690XTzy2UDnrk=";
+    sha256 = "sha256-+ewKbce+rjNWQ0nQzm6O4xSwgzizSPpDPidkQYuoSTU=";
   };
 
   nativeBuildInputs = [ cmake ];
 
   cmakeFlags = [ "-DBUILD_TESTS=ON" ];
 
   doCheck = true;
-  nativeCheckInputs = [ gtest ];
-  checkTarget = "xtest";
-  GTEST_FILTER =
+  nativeCheckInputs = [ doctest ];
+  checkPhase =
     let
       # Upstream Issue: https://github.com/xtensor-stack/xsimd/issues/456
-      filteredTests = lib.optionals stdenv.hostPlatform.isDarwin [
-        "error_gamma_test/*"
+      excludeTestCases = lib.optionals stdenv.hostPlatform.isDarwin [
+        "*gamma*" # The actual test-case name is "[gamma error]"
       ];
+      testCommand = ''./test/test_xsimd --test-case-exclude="${builtins.concatStringsSep "," excludeTestCases}"'';
```

Comment:
I'd rather this were a doctest setup hook or something, but I don't think I'm going to be the one to write it